src/tests/test_delay.cpp

```diff
@@ -0,0 +1,84 @@
+//                   ReactivePlusPlus library
+//
+//           Copyright Aleksey Loginov 2022 - present.
+//                     TC Wang 2022 - present.
+//  Distributed under the Boost Software License, Version 1.0.
+//     (See accompanying file LICENSE_1_0.txt or copy at
+//           https://www.boost.org/LICENSE_1_0.txt)
+//
+//  Project home: https://github.com/victimsnino/ReactivePlusPlus
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <rpp/operators/delay.hpp>
+#include <rpp/schedulers/trampoline_scheduler.hpp>
+#include <rpp/sources/empty.hpp>
+#include <rpp/sources/error.hpp>
+#include <rpp/sources/just.hpp>
+
+#include "mock_observer.hpp"
+
+SCENARIO("delay mirrors both source observable and trigger observable", "[delay]")
+{
+    std::chrono::milliseconds delay_duration{300};
+
+    GIVEN("observable of -1-|")
+    {
+        auto mock = mock_observer<int>{};
+        const auto now = rpp::schedulers::clock_type::now();
+
+        rpp::source::just(1)
+            .delay(delay_duration, rpp::schedulers::trampoline{})
+            .as_blocking()
+            .subscribe(mock);
+
+        THEN("should see -1-| after the delay")
+        {
+            CHECK(rpp::schedulers::clock_type::now() >= now + std::chrono::duration_cast<rpp::schedulers::duration>(delay_duration));
```

Comment:
To be honest, it is better to do this check INSIDE `on_next`. It would be more correct and understandable